packages/plugin-mongo/src/index.ts

```diff
@@ -119,10 +119,11 @@ Database.extend(MongoDatabase, {
   },
 
   async getAssignedChannels(fields, assignMap = this.app.getSelfIds()) {
+    const project = Object.assign({}, projection(fields), projection(['pid', 'type']))
     const channels = await this.channel.find({
       $or: Object.entries(assignMap).map<any>(([type, ids]) => ({ type, assignee: { $in: ids } })),
-    }).project(projection(fields)).toArray()
-    return channels.map(channel => ({ ...pick(Channel.create(channel.type, channel.pid), fields), ...channel, id: `${channel.type}:${channel.pid}` }))
+    }).project(project).toArray()
+    return channels.map(channel => ({ ...pick(Channel.create(channel.type, channel.id), fields), ...channel, id: `${channel.type}:${channel.pid}` }))
```

Comment:
channel.id is platform:xxx and channel.pid is xxx
keep channel.pid here.

Reply:
done